examples/question_answering/question_answering/workflow.py

```diff
@@ -0,0 +1,199 @@
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import dataclasses
+from typing import List
+
+from pydantic.dataclasses import dataclass
+
+from kolena.workflow import define_workflow
+from kolena.workflow import EvaluatorConfiguration
+from kolena.workflow import GroundTruth as BaseGroundTruth
+from kolena.workflow import Inference as BaseInference
+from kolena.workflow import Metadata
+from kolena.workflow import MetricsTestCase
+from kolena.workflow import MetricsTestSample
+from kolena.workflow import MetricsTestSuite
+from kolena.workflow import Text
+from kolena.workflow.annotation import ClassificationLabel
+
+
+@dataclass(frozen=True)
+class TestSample(Text):
+    """Test sample type for the Question Answering workflow."""
+
+    data_id: str
+    """The identifier of the story."""
+
+    question: str
+    """The question based on the story."""
+
+    turn: int
+    """The index of the question asked for the same story."""
+
+    metadata: Metadata = dataclasses.field(default_factory=dict)
+    """The metadata associated with the test sample."""
+
+
+@dataclass(frozen=True)
+class GroundTruth(BaseGroundTruth):
+    """Ground truth type for the Question Answering workflow."""
+
+    answer: str
+    """The human created answer for the question based on the story."""
+
+    clean_answer: str
+    """A normalized and cleaned copy of the answer"""
+
+    context: List[str]
+    """The questions and answers in the history of the conversation."""
+
+    other_answer_1: str
+    """An alternative answer for the question."""
+
+    other_answer_2: str
+    """An alternative answer for the question."""
+
+    other_answer_3: str
+    """An alternative answer for the question."""
+
+    question_answer: ClassificationLabel
+    """A label to display the question and answer with the story."""
+
+
+@dataclass(frozen=True)
+class Inference(BaseInference):
+    """Inference type for the Question Answering workflow."""
+
+    answer: str
+    """The model's generated answer."""
+
+    clean_answer: str
+    """A normalized and cleaned copy of the answer."""
+
+    wc_answer: int
+    """Number of words used in the answer."""
+
+    inference_prompt_tokens: int
+    """Number of tokens used in the prompt."""
+
+    inference_completion_tokens: int
+    """Number of tokens used in the answer."""
+
+    inference_total_tokens: int
+    """Number of tokens used in total."""
+
+    source: str
+    """The location of the stored ground truth data."""
+
+
+workflow, TestCase, TestSuite, Model = define_workflow(
+    "Question Answering",
+    TestSample,
+    GroundTruth,
+    Inference,
+)
+
+
+@dataclass(frozen=True)
+class TestSampleMetrics(MetricsTestSample):
+    """Sample-level metrics for the Question Answering workflow."""
+
+    is_correct: bool
+    """An indication of the answer being correct or wrong."""
```

Comment:
I'm guessing this is based on the threshold specified in the Evaluator Config? Which metric is used to define correctness?  Let's add this to the comment.